### PR TITLE
encode metadata to safe transmit across grpc

### PR DESCRIPTION
**Pull request by vtolstov**
Signed-off-by: Vasiliy Tolstov <v.tolstov@unistack.org>

### CODE REVIEW

File: metadata/metadata.go [hunk 1]
```diff
@@ -47,18 +51,30 @@ func FromContext(ctx context.Context) (Metadata, bool) {
...
 
-	// capitalise all values
-	newMD := make(map[string]string)
+	newMD := make(Metadata)
+
 	for k, v := range md {
-		newMD[strings.Title(k)] = v
+		key, err := base32.StdEncoding.DecodeString(k)
+		if err != nil {
+			panic(err)
```
Comment (asim) [inline metadata/metadata.go]: Should you panic?

Comment (vtolstov) [reply metadata/metadata.go]: sorry, debug stuff that never can happen

File: metadata/metadata.go [hunk 2]
```diff
@@ -47,18 +51,29 @@ func FromContext(ctx context.Context) (Metadata, bool) {
...
 
-	// capitalise all values
-	newMD := make(map[string]string)
+	newMD := make(Metadata)
+
 	for k, v := range md {
-		newMD[strings.Title(k)] = v
+		key, err := base32.StdEncoding.DecodeString(k)
+		if err != nil {
+			return nil, false
```
Comment (asim) [inline metadata/metadata.go]: Should you just continue? What if we're trying to be backwards compatible and there are no base32 encoded strings

Comment (vtolstov) [reply metadata/metadata.go]: inside context we always have encoded metadata

Comment (vtolstov) [reply metadata/metadata.go]: or you mean if we have older micro aff and never?

Comment (vtolstov) [reply metadata/metadata.go]: push fix

File: metadata/metadata.go [hunk 3]
```diff
@@ -47,18 +52,30 @@ func FromContext(ctx context.Context) (Metadata, bool) {
...
-	// capitalise all values
-	newMD := make(map[string]string)
+	newMD := make(Metadata)
+
 	for k, v := range md {
-		newMD[strings.Title(k)] = v
+		key, err := base32.StdEncoding.DecodeString(strings.ToUpper(k))
+		if err != nil {
+			// try to be compatible with older micro versions
+			newMD[strings.Title(k)] = v
```
Comment (asim) [inline metadata/metadata.go:61]: You're not continuing here so it will do this and the thing below

Comment (vtolstov) [reply metadata/metadata.go:61]: inattentive today , thanks

**Comment by asim** [comment]
We will leave for now as this breaks backwards compatibility with existing services.

Encode/Decode methods are welcomed to add as an optional step

**Pull request closed without merging**